firestore/src/main/firestore_main.h

```diff
@@ -107,6 +108,8 @@ class FirestoreInternal {
     return firestore_core_->database_id();
   }
 
+  const std::string& database_name() const { return database_name_; }
```

Comment:
database_name is overloaded here, there is a database_id already, which is a combination of project id and database id/name.

I don't think having a separate database_name_ field is the way to go, we can reuse the existing database_id() from firestore_core_.

Reply:
You can still keep the database_name() method because it looks easier for Android.